libraries/testing/tester.cpp

```diff
@@ -317,6 +317,17 @@ namespace eosio { namespace testing {
          }
       }
 
+      // libtester does not enforce EOSVMOC limits for all tests except those in
+      // eosvmoc_limits_test.cpp, where one or more of the limits are set to
+      // max to indicate the limits are enforced.
+      if (cfg.eosvmoc_config.cpu_limits.rlim_cur != RLIM_INFINITY && cfg.eosvmoc_config.vm_limits.rlim_cur != RLIM_INFINITY && cfg.eosvmoc_config.stack_size_limit != std::numeric_limits<uint64_t>::max() && cfg.eosvmoc_config.generated_code_size_limit != std::numeric_limits<size_t>::max()) {
+         // set to inifinity or max such that they are not enforced.
+         cfg.eosvmoc_config.cpu_limits.rlim_cur       = RLIM_INFINITY;
+         cfg.eosvmoc_config.vm_limits.rlim_cur        = RLIM_INFINITY;
+         cfg.eosvmoc_config.stack_size_limit          = std::numeric_limits<uint64_t>::max();
+         cfg.eosvmoc_config.generated_code_size_limit = std::numeric_limits<size_t>::max();
+      }
```

Comment:
I can't help but feel `std::optional` would be a better choice for these items. `default_config()` could be modified to simply `.reset()` each of them,
https://github.com/AntelopeIO/leap/blob/e4df534c675e8133b257a783cde4326c76618c93/libraries/testing/include/eosio/testing/tester.hpp#L400-L410
and since that occurs before validating_tester's `Lambda conf_edit`, eosvmoc_limits_tests can still override them back to however it wants them.

Regardless of `optional` usage, doing this part in `default_config()` feels like right spot since that's where other libtester-specific configuration items are set.

An optional should be able to be plumbed through easily, and something like
https://github.com/AntelopeIO/leap/blob/e4df534c675e8133b257a783cde4326c76618c93/libraries/chain/webassembly/runtimes/eos-vm-oc/compile_trampoline.cpp#L172-L174
becomes
```c++
if(conf.cpu_limits) {
   struct rlimit cpu_limit = {*conf.cpu_limit, *conf.cpu_limit};
   setrlimit(RLIMIT_CPU, &cpu_limit);
}
```